Sources/OTelCore/OTel+Configuration.swift

```diff
@@ -0,0 +1,259 @@
+            maxExportBatchSize: 512
+        )
+    }
+}
+
+extension OTel.Configuration.TracesConfiguration {
+    public struct ExporterSelection: Sendable {
+        package enum Backing: Sendable {
+            case otlp
+            case jaeger
+            case zipkin
+            case console
+        }
+
+        package var backing: Backing
+
+        public static let otlp: Self = .init(backing: .otlp)
+        public static let jaeger: Self = .init(backing: .jaeger)
+        public static let zipkin: Self = .init(backing: .zipkin)
+        public static let console: Self = .init(backing: .console)
+    }
+}
+
+extension OTel.Configuration.MetricsConfiguration {
+    public struct ExporterSelection: Sendable {
+        package enum Backing: Sendable {
+            case otlp
+            case prometheus
+            case console
+        }
+
+        package var backing: Backing
+
+        public static let otlp: Self = .init(backing: .otlp)
+        public static let prometheus: Self = .init(backing: .prometheus)
+        public static let console: Self = .init(backing: .console)
+    }
+}
+
+extension OTel.Configuration.LogsConfiguration {
+    public struct ExporterSelection: Sendable {
+        package enum Backing: Sendable {
+            case otlp
+            case console
+        }
+
+        package var backing: Backing
+
+        public static let otlp: Self = .init(backing: .otlp)
+        public static let console: Self = .init(backing: .console)
+    }
+}
+
+extension OTel.Configuration {
+    public struct OTLPExporterConfiguration: Sendable {
+        public var endpoint: String
+        public var insecure: Bool
+        public var certificateFilePath: String?
+        public var clientKeyFilePath: String?
+        public var clientCertificateFilePath: String?
+        public var headers: [(String, String)]
+        public var compression: Compression
+        public var timeout: Duration
+        public var `protocol`: Protocol
+
+        public static let `default`: Self = .init(
+            endpoint: "http://localhost:4318",
+            insecure: false,
+            certificateFilePath: nil,
+            clientKeyFilePath: nil,
+            clientCertificateFilePath: nil,
+            headers: [],
+            compression: .none,
+            timeout: .seconds(10),
+            protocol: .httpProtobuf
+        )
+    }
+}
+
+extension OTel.Configuration.OTLPExporterConfiguration {
+    public struct Compression: Sendable {
+        package enum Backing {
+            case gzip
+            case none
+        }
+
+        package var backing: Backing
+
+        public static let none: Self = .init(backing: .none)
+        public static let gzip: Self = .init(backing: .gzip)
+    }
+
+    // swiftformat:disable:redundantBackticks
+    public struct `Protocol`: Equatable, Sendable {
+        package enum Backing {
+            case grpc
+            case httpProtobuf
+            case httpJSON
+        }
```

Comment:
Should these be behind the package traits?

Reply:
Yes. Not these ones, but I've added a fixup commit that adds the unavailability on the public symbols below.